bazel/repositories.bzl

```diff
@@ -19,92 +19,24 @@ WINDOWS_SKIP_TARGETS = [
-
-        if "project_name" not in location:
-            _fail_missing_attribute("project_name", key)
-        mutable_location.pop("project_name")
-
-        if "project_desc" not in location:
-            _fail_missing_attribute("project_desc", key)
-        mutable_location.pop("project_desc")
-
-        if "project_url" not in location:
-            _fail_missing_attribute("project_url", key)
-        project_url = mutable_location.pop("project_url")
-        if not project_url.startswith("https://") and not project_url.startswith("http://"):
-            fail("project_url must start with https:// or http://: " + project_url)
-
-        if "version" not in location:
-            _fail_missing_attribute("version", key)
-        mutable_location.pop("version")
-
-        if "use_category" not in location:
-            _fail_missing_attribute("use_category", key)
-        use_category = mutable_location.pop("use_category")
-
-        if "dataplane_ext" in use_category or "observability_ext" in use_category:
-            if "extensions" not in location:
-                _fail_missing_attribute("extensions", key)
-            mutable_location.pop("extensions")
-
-        if "last_updated" not in location:
-            _fail_missing_attribute("last_updated", key)
-        last_updated = mutable_location.pop("last_updated")
-
-        # Starlark doesn't have regexes.
-        if len(last_updated) != 10 or last_updated[4] != "-" or last_updated[7] != "-":
-            fail("last_updated must match YYYY-DD-MM: " + last_updated)
-
-        if "cpe" in location:
-            cpe = mutable_location.pop("cpe")
-
-            # Starlark doesn't have regexes.
-            cpe_matches = (cpe != "N/A" and (not cpe.startswith("cpe:2.3:a:") or not cpe.endswith(":*") and len(cpe.split(":")) != 6))
-            if cpe_matches:
-                fail("CPE must match cpe:2.3:a:<facet>:<facet>:*: " + cpe)
-        elif not [category for category in USE_CATEGORIES_WITH_CPE_OPTIONAL if category in location["use_category"]]:
-            _fail_missing_attribute("cpe", key)
-
-        for category in location["use_category"]:
-            if category not in USE_CATEGORIES:
-                fail("Unknown use_category value '" + category + "' for dependecy " + key)
-
-    return locations
-
-REPOSITORY_LOCATIONS = _repository_locations()
-
-# To initialize http_archive REPOSITORY_LOCATIONS dictionaries must be stripped of annotations.
-# See repository_locations.bzl for the list of annotation attributes.
-def _get_location(dependency):
-    stripped = dict(REPOSITORY_LOCATIONS[dependency])
-    for attribute in DEPENDENCY_ANNOTATIONS:
-        stripped.pop(attribute, None)
-    return stripped
-
-def _repository_impl(name, **kwargs):
+REPOSITORY_LOCATIONS = load_repository_locations(REPOSITORY_LOCATIONS_SPEC)
+
+# Use this macro to reference any HTTP archive from bazel/repository_locations.bzl.
+def external_http_archive(name, **kwargs):
```

Comment:
why  not put this wrapper in a file to share between API and this file?
(does `envoy_http_archive` ever have a use, or can this co-opt that?)

Reply:
This essentially wraps up combining the file-local `REPOSITORY_LOCATIONS` with `http_archive`. If moved outside, it would still need something local to do this with whatever external macro was defined.

Reply:
argh, thanks! i missed that.